src/Selenium2Library/__init__.py

```diff
@@ -184,10 +183,42 @@ def __init__(self,
         | Library `|` Selenium2Library `|` implicit_wait=5 `|` run_on_failure=Log Source | # Sets default implicit_wait to 5 seconds and runs `Log Source` on failure |
         | Library `|` Selenium2Library `|` timeout=10      `|` run_on_failure=Nothing    | # Sets default timeout to 10 seconds and does nothing on failure           |
         """
-        for base in Selenium2Library.__bases__:
-            base.__init__(self)
+        self.logger = logger
+        libraries = [
+            AlertKeywords(self),
+            BrowserManagementKeywords(self),
+            RunOnFailureKeywords(self),
+            ElementKeywords(self),
+            TableElementKeywords(self),
+            FormElementKeywords(self),
+            SelectElementKeywords(self),
+            JavaScriptKeywords(self),
+            CookieKeywords(self),
+            ScreenshotKeywords(self),
+            WaitingKeywords(self)
+        ]
+
+        DynamicCore.__init__(self, libraries)
+        self.cache = BrowserCache()
+
         self.screenshot_root_directory = screenshot_root_directory
-        self.set_selenium_timeout(timeout)
-        self.set_selenium_implicit_wait(implicit_wait)
-        self.register_keyword_to_run_on_failure(run_on_failure)
+        # self.set_selenium_timeout(timeout)
+        # self.set_selenium_implicit_wait(implicit_wait)
+        # self.register_keyword_to_run_on_failure(run_on_failure)
         self.ROBOT_LIBRARY_LISTENER = LibraryListener()
+
+    def run_keyword(self, name, args, kwargs):
+        try:
+            return DynamicCore.run_keyword(self, name, args, kwargs)
+        except Exception as e:
+            print 'run failure here'
+            raise e
+
+    def current_browser(self):
```

Comment:
A hundred times yes. Please leave `_current_browser()` for a release cycle as it will break a whole lot of outside code. I suggest emitting a warning every time it is invoked. Or perhaps only once.
There are many aspects to Selenium S2L does not expose so a lot of people depend on this private method. In the mean time you can have `_current_browser()` call `browser` or `current_browser()`.

I also agree it would be better suited as an attribute rather than a method.